ledger/src/blockstore/blockstore_purge.rs

```diff
@@ -235,16 +265,90 @@ impl Blockstore {
             }
         }
         delete_range_timer.stop();
+        purge_stats.delete_range += delete_range_timer.as_us();
+
+        Ok(())
+    }
+
+    /// Purge coding shreds and clear coding shred indices for [`from_slot`, `to_slot`].
+    fn run_purge_coding_shreds(
+        &self,
+        write_batch: &mut WriteBatch,
+        columns_purged: &mut bool,
+        from_slot: Slot,
+        to_slot: Slot,
+        purge_stats: &mut PurgeStats,
+    ) -> Result<()> {
+        // delete_range_cf is not inclusive on the end of range
+        let to_slot = to_slot.saturating_add(1);
+
+        let mut delete_range_timer = Measure::start("delete_range_coding");
+        *columns_purged &= self
+            .db
+            .delete_range_cf::<cf::ShredCode>(write_batch, from_slot, to_slot)
+            .is_ok();
+
+        for (slot, mut index) in self.slot_index_iterator(from_slot)? {
+            if slot >= to_slot {
+                break;
+            }
+            index.coding_mut().clear();
+            *columns_purged &= write_batch.put::<cf::Index>(slot, &index).is_ok();
+        }
```

Comment:
Once a slot has been rooted, I don't think we try to access the coding shreds again. So, we could possibly get away with not updating the index, but that seems like it'd be begging for a bug down the road